### PR TITLE
if event is pull_request, use the config file from head branch of PR

### PR DESCRIPTION
**Pull request by shine2lay**
Goal
---
When an Event is Pull_request, use the head branch's config file instead of the master branches

Context
---
This will allow user to test different settings and see how it works before merging it into the code base

### CODE REVIEW

File: lib/configuration/configuration.js [hunk 1]
```diff
@@ -112,15 +112,29 @@ class Configuration {
...
-      repo: repo.repo,
-      path: Configuration.FILE_NAME
-    }).then(res => {
+    if (context.event === 'pull_request') {
+      return github.repos.getContent({
+        owner: repo.owner,
+        repo: repo.repo,
+        path: Configuration.FILE_NAME,
+        ref: context.payload.pull_request.head.ref
+      })
```
Comment (jusx) [inline lib/configuration/configuration.js]: What happens here when there is no Configuration changes or additions in the PR? I think in that scenario it still needs to retrieve from the repo "master"

**Comment by jusx** [comment]
@shine2lay We should probably also update the unit tests.

**Comment by shine2lay** [comment]
@jusx updated

File: lib/configuration/configuration.js (hunk 1 above)
Comment (jusx) [reply lib/configuration/configuration.js]: OK sounds good. If that's how the `github.repos.getContent` we are gtg.

**Pull request merged**